src/toil/batchSystems/singleMachine.py

```diff
@@ -241,15 +262,27 @@ def __init__(self, startTime, popen, killIntended):
 
 
 class ResourcePool(object):
-    def __init__(self, initial_value):
+    def __init__(self, initial_value, resourceType):
         super(ResourcePool, self).__init__()
         self.condition = Condition()
         self.value = initial_value
+        self.resourceType = resourceType
 
     def acquire(self, amount):
         with self.condition:
+            startTime = time.time()
             while amount > self.value:
-                self.condition.wait()
+                if time.time() - startTime >= 5:
+                    # This means the thread timed out waiting for the resource.  We exit the nested
+                    # context managers in worker to prevent blocking of a resource due to
+                    # unavailability of a nested resource request.
+                    raise InsufficientSystemResources(resource=self.resourceType, requested=amount,
```

Comment:
This is not the right exception type. Create new one, nest it in ResourcePool and call it `AcquisitionTimeoutException`.